wallet/wallet_test.go

```diff
@@ -1946,11 +1946,16 @@ func TestV2TPoolRace(t *testing.T) {
 	// output in the spend transaction invalid unless it is updated.
 	mineAndSync(t, cm, ws, w, types.VoidAddress, 1)
 
-	// broadcast the transaction set including the already confirmed setup
-	// transaction. This seems unnecessary, but it's a fairly common occurrence
-	// when passing transaction sets using unconfirmed outputs between a renter
-	// and host. If the transaction set is not updated correctly, it will fail.
-	if _, err := cm.AddV2PoolTransactions(basis, []types.V2Transaction{setupTxn, spendTxn}); err != nil {
+	// now that the setup transaction has been accepted, we can't include it in
+	// the set anymore, or we'll get a double-spend error. AddV2PoolTransactions
+	// will internally replace the ephemeral output, making spendTxn valid.
+	if _, err := cm.AddV2PoolTransactions(basis, []types.V2Transaction{spendTxn}); err != nil {
```

Comment:
Should extend this with a case that broadcasts both and checks the returned error for a double spend.

Reply:
+1

Reply:
huh, apparently `AddV2PoolTransactions` will also silently skip any transactions that were confirmed since `basis`. Definitely convenient, but it does feel a bit weird that we don't do it for v1 transactions too. 🤔

Reply:
That behavior is useful for rebroadcasting contract formation sets on the host

Reply:
Avoiding the race is a pretty big benefit of keeping them combined. Basis and proof updates in RHP4 have enough complexity without adding another bit 😓 

I'm not sure if updating `AddPoolTransactions` to include a basis is worth it. v1 isn't equipped to pass it around, unlike v2 where we explicitly store basis alongside the transaction set.